### PR TITLE
Set current key for custom multi option fields for profiles

### LINKED ISSUE #4451: Gender Field
Last Pull request: 
https://github.com/SimpleMachines/SMF2.1/pull/4445

Forum version: SMF 2.1 Beta 4 (more detailed)
Current SMF version: SMF 2.1 Beta 3
GD version: bundled (2.1.0 compatible) 
MySQLi engine: MariaDB 
MySQLi version: 10.1.25-MariaDB 
Alternative PHP Cache: 5.1.2 
PHP: 7.1.11 (more detailed) 
Server version: Apache/2.4.29 (Win32) OpenSSL/1.0.2l PHP/7.1.11 
Fix: 
https://github.com/SimpleMachines/SMF2.1/pull/4383

Fix not work with the new pull request today. Not by new install or update. Key is not read from the database. This field is not displayed in the post view.

gender_2 

gender_1 

gender_3 

gender_4

### PR DESCRIPTION
**Pull request by equinoxmatt**
Fixes #4451

Took a very similar approach to how loadMemberContext() does it for member profiles.  Same concept, check all the custom fields on the profile being loaded to see if they have options, match the option to a profile option, check if the values match and set a key.

Then added {KEY} to the output_html.

### CODE REVIEW

File: Sources/Profile.php [hunk 1]
```diff
@@ -886,6 +886,17 @@ function loadCustomFields($memID, $area = 'summary')
 	$context['custom_fields_required'] = false;
 	while ($row = $smcFunc['db_fetch_assoc']($request))
 	{
+		$value = $user_profile[$memID]['options'][$row['col_name']];
+		$currentKey = 0;
+		if (!empty($row['field_options'])) {
```
Comment (Sesquipedalian) [inline Sources/Profile.php]: Nits: 
1. Please move opening braces to a new line.
2. Please remove braces entirely for the `if (empty($currentKey))` block.

**Review by Sesquipedalian — CHANGES_REQUESTED** [review]
LGTM, apart from some code formatting.

**Comment by Sesquipedalian** [comment]
@winstar [reports](https://github.com/SimpleMachines/SMF2.1/issues/4589#issuecomment-371233175) that the gender icon display isn't working in the member list either. Could you please add some code to deal with that, @equinoxmatt?

**Comment by equinoxmatt** [comment]
@Sesquipedalian  will clear this up over the weekend.

File: Sources/Profile.php (hunk 1 above)
Comment (equinoxmatt) [reply Sources/Profile.php]: Arrghh, feels so wrong, but ok :)

Comment (Sesquipedalian) [reply Sources/Profile.php]: In SMF 3, I'd like to change the coding style for everything. But that's not where we are now.

**Comment by equinoxmatt** [comment]
Didn't get this done over the weekend, I might be able to get onto it later this week 👍

**Comment by equinoxmatt** [comment]
@Sesquipedalian  Fixed member list and formatting can you review please.

**Comment by wintstar** [comment]
Sorry, I just received an error message.
>  Type of error: Undefined
> 8: Undefined index: cust_gender
>  http://localhost/smfbeta/index.php?action=mlist
>  C:/xampp/htdocs/smfbeta/Sources/Memberlist.php (Line 635)

File: Sources/Memberlist.php [hunk 2]
```diff
@@ -631,6 +631,19 @@ function printMemberListRows($request)
 		{
 			foreach ($context['custom_profile_fields']['columns'] as $key => $column)
 			{
+
```
Comment (Sesquipedalian) [inline Sources/Memberlist.php:634]: Nit: Unnecessary blank line.

**Review by Sesquipedalian — CHANGES_REQUESTED** [review]
One nit. Otherwise, LGTM.

**Comment by Sesquipedalian** [comment]
Meh. I'll just merge this and then remove the blank line myself.

**Comment by wintstar** [comment]
@Sesquipedalian This fix only causes errors, I reported them before you released this pull request #4589

> wintstar at 17.03.2018  09:57
> Sorry, I just received an error message.
>
> Type of error: Undefined
> 8: Undefined index: cust_gender
> http://localhost/smfbeta/index.php?action=mlist
> C:/xampp/htdocs/smfbeta/Sources/Memberlist.php (Line 635)

**Pull request merged**